README.md
<img src="img/header.png" alt="Cinder | A clean, responsive theme for MkDocs" width="728">

## Cinder

A clean, responsive MkDocs static documentation site generator theme

[Demo and Documentation](http://sourcefoundry.org/cinder/)



CHANGELOG.md
## Changelog

### v0.17.0

- added support for keyboard shortcut bindings (PR #71)
- updated documentation with keyboard shortcut binding settings in the theme yaml settings file
- fixed: incorrect border radius CSS settings in dropdown menus (PR #72)
- fixed: webfont.js URL (PR #69)

### v0.16.3

- fixed: Hack webfont URL

### v0.16.2

- fixed: missing 404 page build (PR #65)
- fixed: focus search input when opening search modal (PR #63)

### v0.16.1

- included page revision metadata in the footer block of the base.html template file

### v0.16.0

- fix for jQuery associated bug that lead to failure of dropdown menu behavior as of v0.15.0. This required us to revert to v1.12.4 of jQuery as is used with Bootstrap 3 theme
- removed embedded jQuery and transitioned to jQuery served from Google CDN

### v0.15.0

- BACKWARDS INCOMPATIBLE UPDATE - you will need to update your Font Awesome icon classes, possibly make other changes.  Read through the modifications in detail and test your site if you are upgrading from previous versions of the Cinder theme.
- added admonition rendering support for note, warning, danger types from Markdown markup
- added Bootstrap alert support
- added Bootstrap-style callout support (derived and slightly modified from Chris Pratt's CSS source under MIT License https://codepen.io/chrisdpratt/details/IAymB/)
- added modified Github syntax highlighter CSS as default
- updated Font Awesome to v5.5.0
- updated highlightjs to v9.13.1
- updated jQuery to v3.3.1
- updated html5shiv to v3.7.3
- updated respond.js to v1.4.2
- updated Google webfont loader (webfont.js) to v1.6.28
- updated Hack typeface to v3.003 release
- fixed highlightjs syntax highlighter support
- modified highlightjs syntax highlighter language support to 29 languages
- fixed ToC scrolling when height > window height (Issue #43, PR #55)
- updated install/usage documentation with MkDocs v1.0+ YAML syntax
- updated install/usage documentation with information about new optional blocks
- updated specimen documentation with new source code example
- updated specimen documentation with new alert examples
- updated specimen documentation with new callouts examples
- removed `sites` directory from git version control
- minified all CSS files used in the theme
- added .gitignore targets

### v0.14.0

- added htmltitle block
- added extrahead block
- added footer block
- added optional revision date metadata in the footer

### v0.13.0

- added option to disable table of contents in a page-specific manner with page metadata setting `disable_toc: true`
- bugfix: fixed Google Analytics JavaScript in builds when this is defined in settings

### v0.12.3

- bugfix: fixed navigation links to address changes that were released in mkdocs v1.0

### v0.12.2

- bugfix: added mock setting to the `mkdocs_theme.yml` file to address an upstream mkdocs bug that occurs when this file is blank (issue #46)

### v0.12.1

- bugfix: added `*.yml` to MANIFEST.in file to support new yml settings file required for mkdocs v1.0+

### v0.12.0

- added `mkdocs_theme.yml` file to support mkdocs v1.0 settings file changes

### v0.11.0

- added support for Gitlab icon in navigation menus

### v0.10.2

- Fix the templates in base.html that have been changed in mkdocs 0.17

### v0.10.1

- added responsive image support

### v0.10.0

- added support for MkDocs v0.17 changes
- updated FontAwesome to v4.7.0

### v0.9.4

- added support for H3 header titles in sidebar menu (PR #9)

### v0.9.3

- Added support for > 2 level dropdown link depth (Issue #1 with PR #3)
- Updated Hack web fonts to v2.018

### v0.9.2

- Updated Hack web fonts to v2.015.


### v0.9.1

- modified the width of the footer horizontal line to avoid overlap with long TOC in the left sidebar
- minified the Bootstrap CSS
- modified the formatting of license declaration displays in the footer

### v0.9.0

- Initial release


cinder/js/base.js
function getSearchTerm()
{
    var sPageURL = window.location.search.substring(1);
    var sURLVariables = sPageURL.split('&');
    for (var i = 0; i < sURLVariables.length; i++)
    {
        var sParameterName = sURLVariables[i].split('=');
        if (sParameterName[0] == 'q')
        {
            return sParameterName[1];
        }
    }
}

$(document).ready(function() {
    /**
     * ------------------------------------------------------------------------
     * Cinder theme specific
     * ------------------------------------------------------------------------
     */
    hljs.initHighlightingOnLoad();


    /**
     * ------------------------------------------------------------------------
     * Taken from themes/mkdocs/js/base.js
     * ------------------------------------------------------------------------
     */
    var search_term = getSearchTerm(),
    $search_modal = $('#mkdocs_search_modal'),
    $keyboard_modal = $('#mkdocs_keyboard_modal');

    if(search_term){
        $search_modal.modal();
    }

    // make sure search input gets autofocus everytime modal opens.
    $search_modal.on('shown.bs.modal', function () {
        $search_modal.find('#mkdocs-search-query').focus();
    });

    // Close search modal when result is selected
    // The links get added later so listen to parent
    $('#mkdocs-search-results').click(function(e) {
      if ($(e.target).is('a')) {
        $search_modal.modal('hide');
      }
    });

    if (typeof shortcuts !== 'undefined') {
      // Populate keyboard modal with proper Keys
      $keyboard_modal.find('.help.shortcut kbd')[0].innerHTML = keyCodes[shortcuts.help];
      $keyboard_modal.find('.prev.shortcut kbd')[0].innerHTML = keyCodes[shortcuts.previous];
      $keyboard_modal.find('.next.shortcut kbd')[0].innerHTML = keyCodes[shortcuts.next];
      $keyboard_modal.find('.search.shortcut kbd')[0].innerHTML = keyCodes[shortcuts.search];

      // Keyboard navigation
      document.addEventListener("keydown", function(e) {
          if ($(e.target).is(':input')) return true;
          var key = e.which || e.keyCode || window.event && window.event.keyCode;
          var page;
          switch (key) {
              case shortcuts.next:
                  page = $('.navbar a[rel="next"]:first').prop('href');
                  break;
              case shortcuts.previous:
                  page = $('.navbar a[rel="prev"]:first').prop('href');
                  break;
              case shortcuts.search:
                  e.preventDefault();
                  $keyboard_modal.modal('hide');
                  $search_modal.modal('show');
                  $search_modal.find('#mkdocs-search-query').focus();
                  break;
              case shortcuts.help:
                  $search_modal.modal('hide');
                  $keyboard_modal.modal('show');
                  break;
              default: break;
          }
          if (page) {
              $keyboard_modal.modal('hide');
              window.location.href = page;
          }
      });
    }

    $('table').addClass('table table-striped table-hover');

    // Improve the scrollspy behaviour when users click on a TOC item.
    $(".bs-sidenav a").on("click", function() {
        var clicked = this;
        setTimeout(function() {
            var active = $('.nav li.active a');
            active = active[active.length - 1];
            if (clicked !== active) {
                $(active).parent().removeClass("active");
                $(clicked).parent().addClass("active");
            }
        }, 50);
    });
});


/**
 * ------------------------------------------------------------------------
 * Taken from themes/mkdocs/js/base.js
 * ------------------------------------------------------------------------
 */

$('body').scrollspy({
    target: '.bs-sidebar',
    offset: 100
});

/* Prevent disabled links from causing a page reload */
$("li.disabled a").click(function() {
    event.preventDefault();
});

// See https://www.cambiaresearch.com/articles/15/javascript-char-codes-key-codes
// We only list common keys below. Obscure keys are omited and their use is discouraged.
var keyCodes = {
  8: 'backspace',
  9: 'tab',
  13: 'enter',
  16: 'shift',
  17: 'ctrl',
  18: 'alt',
  19: 'pause/break',
  20: 'caps lock',
  27: 'escape',
  32: 'spacebar',
  33: 'page up',
  34: 'page down',
  35: 'end',
  36: 'home',
  37: '&larr;',
  38: '&uarr;',
  39: '&rarr;',
  40: '&darr;',
  45: 'insert',
  46: 'delete',
  48: '0',
  49: '1',
  50: '2',
  51: '3',
  52: '4',
  53: '5',
  54: '6',
  55: '7',
  56: '8',
  57: '9',
  65: 'a',
  66: 'b',
  67: 'c',
  68: 'd',
  69: 'e',
  70: 'f',
  71: 'g',
  72: 'h',
  73: 'i',
  74: 'j',
  75: 'k',
  76: 'l',
  77: 'm',
  78: 'n',
  79: 'o',
  80: 'p',
  81: 'q',
  82: 'r',
  83: 's',
  84: 't',
  85: 'u',
  86: 'v',
  87: 'w',
  88: 'x',
  89: 'y',
  90: 'z',
  91: 'Left Windows Key / Left ⌘',
  92: 'Right Windows Key',
  93: 'Windows Menu / Right ⌘',
  96: 'numpad 0',
  97: 'numpad 1',
  98: 'numpad 2',
  99: 'numpad 3',
  100: 'numpad 4',
  101: 'numpad 5',
  102: 'numpad 6',
  103: 'numpad 7',
  104: 'numpad 8',
  105: 'numpad 9',
  106: 'multiply',
  107: 'add',
  109: 'subtract',
  110: 'decimal point',
  111: 'divide',
  112: 'f1',
  113: 'f2',
  114: 'f3',
  115: 'f4',
  116: 'f5',
  117: 'f6',
  118: 'f7',
  119: 'f8',
  120: 'f9',
  121: 'f10',
  122: 'f11',
  123: 'f12',
  124: 'f13',
  125: 'f14',
  126: 'f15',
  127: 'f16',
  128: 'f17',
  129: 'f18',
  130: 'f19',
  131: 'f20',
  132: 'f21',
  133: 'f22',
  134: 'f23',
  135: 'f24',
  144: 'num lock',
  145: 'scroll lock',
  186: '&semi;',
  187: '&equals;',
  188: '&comma;',
  189: '&hyphen;',
  190: '&period;',
  191: '&quest;',
  192: '&grave;',
  219: '&lsqb;',
  220: '&bsol;',
  221: '&rsqb;',
  222: '&apos;',
};